src/trait_group.h

```diff
@@ -0,0 +1,202 @@
+#pragma once
+#ifndef TRAIT_GROUP_H
+#define TRAIT_GROUP_H
+
+#include "json.h"
+#include "mutation.h"
+#include "string.h"
+#include "string_id.h"
+
+typedef std::string Trait_group_tag;
```

Comment:
This should be a `string_id` of some kind. Item groups use raw `std::string` because they have not yet been changed to use `string_id`, but `string_id` is preferable as provides a common interface and more type safety.